vcr/holder/oidc4vci_wallet.go

```diff
@@ -158,8 +167,13 @@ func (h wallet) HandleCredentialOffer(ctx context.Context, offer oidc4vci.Creden
 			StatusCode: http.StatusInternalServerError,
 		}
 	}
-	// TODO: Wallet should make sure the VC is of the expected type
-	//       See https://github.com/nuts-foundation/nuts-node/issues/2050
+	if err = credentialTypesMatchOffer(h.jsonldReader, *credential, offer.Credentials[0]); err != nil {
```

Comment:
Should you expand `offer.Credentials[0]` as well before processing it?

Reply:
Why do you prefer to do it there? It is currently expanded inside the function as it is a detail from the format we are using.

Reply:
Nvm, I misread